File: mobile/iosApp/iosApp/UI/Screens/NewsletterDetailScreen.swift
import SwiftUI
import Combine

struct NewsletterDetailScreen: View {
    let newsletterId: String
    @ObservedObject private var viewModel: NewsletterDetailViewModel
    @EnvironmentObject private var navigationState: NavigationState
    @State private var showPreferencesSheet = false
    
    init(newsletterId: String, viewModel: NewsletterDetailViewModel) {
        self.newsletterId = newsletterId
        self.viewModel = viewModel
    }
    
    var body: some View {
        VStack(spacing: AppSpacing.Medium) {
            // Header
            HStack {
                Button("Back") {
                    navigationState.navigateBack()
                }
                .buttonStyle(TextButtonStyle())
                
                Spacer()
                
                Text(AppStrings.Newsletter_Detail)
                    .font(.largeTitle)
                    .fontWeight(.bold)
                    .foregroundColor(Color.primary)
                
                Spacer()
                
                // Action menu would go here
            }
            
            // Content based on state
            switch viewModel.state {
            case .loading:
                LoadingView(style: .circular)
                    .frame(maxWidth: .infinity, maxHeight: .infinity)
                
            case .error(let message):
                ErrorView(
                    message: message,
                    onRetry: { viewModel.loadNewsletter(newsletterId: newsletterId) }
                )
                .frame(maxWidth: .infinity, maxHeight: .infinity)
                
            case .success(let newsletter):
                ScrollView {
                    VStack(spacing: AppSpacing.Medium) {
                        NewsletterHeader(
                            newsletter: newsletter,
                            onSubscribeClick: { viewModel.toggleSubscription() }
                        )
                        
                        AnalyticsSection(
                            newsletter: newsletter,
                            onAnalyticsClick: { /* Navigate to analytics */ }
                        )
                        
                        PerformanceChart(
                            title: "Performance Over Time",
                            data: newsletter.performanceData,
                            chartType: .line
                        )
                        
                        PreferencesSection(
                            onPreferencesClick: { showPreferencesSheet = true }
                        )
                    }
                    .padding(.horizontal, AppSpacing.Medium)
                }
            }
        }
        .padding(AppSpacing.Medium)
        .sheet(isPresented: $showPreferencesSheet) {
            if case .success(let newsletter) = viewModel.state {
                PreferencesSheet(
                    newsletter: newsletter,
                    onDismiss: { showPreferencesSheet = false },
                    onSave: { preferences in
                        viewModel.updatePreferences(preferences)
                        showPreferencesSheet = false
                    }
                )
            }
        }
        .onAppear {
            viewModel.loadNewsletter(newsletterId: newsletterId)
        }
    }
}

struct NewsletterHeader: View {
    let newsletter: NewsletterDetail
    let onSubscribeClick: () -> Void
    
    var body: some View {
        AppCard {
            VStack(alignment: .leading, spacing: AppSpacing.Medium) {
                HStack(alignment: .top) {
                    VStack(alignment: .leading, spacing: AppSpacing.Small) {
                        Text(newsletter.title)
                            .font(.title2)
                            .fontWeight(.semibold)
                            .foregroundColor(Color.primaryText)
                        
                        Text(newsletter.author)
                            .font(.body)
                            .foregroundColor(Color.secondaryText)
                    }
                    
                    Spacer()
                    
                    AppIcon(name: newsletter.icon, size: .large, tint: Color.accent)
                }
                
                Text(newsletter.description)
                    .font(.body)
                    .foregroundColor(Color.secondaryText)
                    .multilineTextAlignment(.leading)
                
                AppButton(
                    title: newsletter.isSubscribed ? "Unsubscribe" : "Subscribe",
                    variant: newsletter.isSubscribed ? .outlined : .primary,
                    action: onSubscribeClick
                )
            }
            .padding(AppSpacing.Medium)
        }
    }
}

struct AnalyticsSection: View {
    let newsletter: NewsletterDetail
    let onAnalyticsClick: () -> Void
    
    var body: some View {
        AppCard {
            VStack(alignment: .leading, spacing: AppSpacing.Medium) {
                HStack {
                    Text("Analytics")
                        .font(.title3)
                        .fontWeight(.semibold)
                        .foregroundColor(Color.primaryText)
                    
                    Spacer()
                    
                    Button("View Details") {
                        onAnalyticsClick()
                    }
                    .buttonStyle(TextButtonStyle())
                }
                
                // Analytics Chips
                VStack(spacing: AppSpacing.Small) {
                    HStack(spacing: AppSpacing.Small) {
                        AnalyticsChip(
                            label: "Subscribers",
                            value: String(newsletter.subscribers),
                            icon: AppIcon.IconProfile
                        )
                        
                        AnalyticsChip(
                            label: "Open Rate",
                            value: String(format: "%.1f%%", newsletter.openRate),
                            icon: AppIcon.IconAnalytics
                        )
                    }
                    
                    HStack(spacing: AppSpacing.Small) {
                        AnalyticsChip(
                            label: "Click Rate",
                            value: String(format: "%.1f%%", newsletter.clickRate),
                            icon: AppIcon.IconAnalytics
                        )
                        
                        AnalyticsChip(
                            label: "Frequency",
                            value: newsletter.frequency,
                            icon: AppIcon.IconCalendar
                        )
                    }
                }
            }
            .padding(AppSpacing.Medium)
        }
    }
}

struct AnalyticsChip: View {
    let label: String
    let value: String
    let icon: String
    
    var body: some View {
        HStack(spacing: AppSpacing.XSmall) {
            AppIcon(name: icon, size: .small, tint: Color.accent)
            
            VStack(alignment: .leading, spacing: 2) {
                Text(value)
                    .font(.caption)
                    .fontWeight(.medium)
                    .foregroundColor(Color.primaryText)
                
                Text(label)
                    .font(.caption2)
                    .foregroundColor(Color.secondaryText)
            }
        }
        .padding(.horizontal, AppSpacing.Small)
        .padding(.vertical, AppSpacing.XSmall)
        .background(Color.surfaceVariant)
        .cornerRadius(AppCornerRadius.Small)
    }
}

struct PerformanceChart: View {
    let title: String
    let data: [NewsletterDetail.PerformancePoint]
    let chartType: ChartType
    
    var body: some View {
        AppCard {
            VStack(alignment: .leading, spacing: AppSpacing.Medium) {
                Text(title)
                    .font(.headline)
                    .fontWeight(.semibold)
                    .foregroundColor(Color.primaryText)
                
                // Chart placeholder
                RoundedRectangle(cornerRadius: AppCornerRadius.Medium)
                    .fill(Color.surfaceVariant)
                    .frame(height: AppMetrics.ChartMinHeight)
                    .overlay(
                        Text("Chart: \(title) (\(data.count) points)")
                            .font(.caption)
                            .foregroundColor(Color.secondaryText)
                    )
            }
            .padding(AppSpacing.Medium)
        }
    }
}

struct PreferencesSection: View {
    let onPreferencesClick: () -> Void
    
    var body: some View {
        AppCard {
            HStack {
                VStack(alignment: .leading, spacing: AppSpacing.XSmall) {
                    Text("Preferences")
                        .font(.headline)
                        .fontWeight(.semibold)
                        .foregroundColor(Color.primaryText)
                    
                    Text("Customize your subscription settings")
                        .font(.caption)
                        .foregroundColor(Color.secondaryText)
                }
                
                Spacer()
                
                Button("Manage") {
                    onPreferencesClick()
                }
                .buttonStyle(OutlinedButtonStyle())
            }
            .padding(AppSpacing.Medium)
        }
    }
}

struct PreferencesSheet: View {
    let newsletter: NewsletterDetail
    let onDismiss: () -> Void
    let onSave: (NewsletterDetail.Preferences) -> Void
    
    @State private var frequency: String
    @State private var notifications: Bool
    @State private var categories: [String]
    
    init(newsletter: NewsletterDetail, onDismiss: @escaping () -> Void, onSave: @escaping (NewsletterDetail.Preferences) -> Void) {
        self.newsletter = newsletter
        self.onDismiss = onDismiss
        self.onSave = onSave
        self._frequency = State(initialValue: newsletter.preferences.frequency)
        self._notifications = State(initialValue: newsletter.preferences.notifications)
        self._categories = State(initialValue: newsletter.preferences.categories)
    }
    
    var body: some View {
        NavigationView {
            Form {
                Section(header: Text("Frequency")) {
                    Picker("Frequency", selection: $frequency) {
                        Text("Daily").tag("daily")
                        Text("Weekly").tag("weekly")
                        Text("Monthly").tag("monthly")
                    }
                    .pickerStyle(SegmentedPickerStyle())
                }
                
                Section {
                    Toggle("Notifications", isOn: $notifications)
                }
                
                Section(header: Text("Categories")) {
                    Text("Select categories to follow")
                        .font(.caption)
                        .foregroundColor(Color.secondaryText)
                }
            }
            .navigationTitle("Subscription Preferences")
            .navigationBarTitleDisplayMode(.inline)
            .toolbar {
                ToolbarItem(placement: .navigationBarLeading) {
                    Button("Cancel") {
                        onDismiss()
                    }
                    .buttonStyle(TextButtonStyle())
                }
                
                ToolbarItem(placement: .navigationBarTrailing) {
                    Button("Save") {
                        let newPreferences = NewsletterDetail.Preferences(
                            frequency: frequency,
                            notifications: notifications,
                            categories: categories
                        )
                        onSave(newPreferences)
                    }
                    .buttonStyle(TextButtonStyle())
                }
            }
        }
    }
}

enum ChartType {
    case line, bar, area, pie
}

struct NewsletterDetailScreen_Previews: PreviewProvider {
    static var previews: some View {
        let viewModel = NewsletterDetailViewModel()
        NewsletterDetailScreen(
            newsletterId: "1",
            viewModel: viewModel
        )
        .environmentObject(NavigationState())
    }
}

File: mobile/androidApp/src/main/java/com/piper/newsletter/android/ui/screens/NewsletterDetailScreen.kt
package com.piper.newsletter.android.ui.screens

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.navigation.NavController
import com.piper.newsletter.android.ui.components.*
import com.piper.newsletter.android.ui.theme.*
import com.piper.newsletter.shared.models.NewsletterDetail
import com.piper.newsletter.shared.viewmodels.NewsletterDetailViewModel
import com.piper.newsletter.shared.viewmodels.NewsletterDetailViewModel.State
import org.koin.androidx.compose.koinViewModel

@Composable
fun NewsletterDetailScreen(
    newsletterId: String,
    navController: NavController,
    viewModel: NewsletterDetailViewModel = koinViewModel()
) {
    val state by viewModel.state.collectAsState()
    var showPreferencesSheet by remember { mutableStateOf(false) }
    
    LaunchedEffect(newsletterId) {
        viewModel.loadNewsletter(newsletterId)
    }
    
    Column(
        modifier = Modifier
            .fillMaxSize()
            .padding(AppSpacing.Medium.dp)
    ) {
        // Header
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically
        ) {
            AppButton(
                onClick = { navController.navigateUp() },
                variant = ButtonVariant.Text,
                text = "Back"
            )
            
            Text(
                text = AppStrings.Newsletter_Detail,
                style = MaterialTheme.typography.headlineLarge,
                color = MaterialTheme.colorScheme.onSurface
            )
            
            // Action menu would go here
        }
        
        Spacer(modifier = Modifier.height(AppSpacing.Large.dp))
        
        // Content based on state
        when (val currentState = state) {
            is State.Loading -> {
                LoadingIndicator(
                    style = LoadingStyle.Circular,
                    modifier = Modifier.fillMaxSize()
                )
            }
            
            is State.Error -> {
                ErrorState(
                    message = currentState.message,
                    onRetry = { viewModel.loadNewsletter(newsletterId) },
                    modifier = Modifier.fillMaxSize()
                )
            }
            
            is State.Success -> {
                NewsletterDetailContent(
                    newsletter = currentState.newsletter,
                    onSubscribeClick = { viewModel.toggleSubscription() },
                    onPreferencesClick = { showPreferencesSheet = true },
                    onAnalyticsClick = { /* Navigate to analytics */ }
                )
            }
        }
    }
    
    // Preferences Sheet
    if (showPreferencesSheet) {
        PreferencesSheet(
            newsletter = (state as? State.Success)?.newsletter,
            onDismiss = { showPreferencesSheet = false },
            onSave = { preferences ->
                viewModel.updatePreferences(preferences)
                showPreferencesSheet = false
            }
        )
    }
}

@Composable
fun NewsletterDetailContent(
    newsletter: NewsletterDetail,
    onSubscribeClick: () -> Unit,
    onPreferencesClick: () -> Unit,
    onAnalyticsClick: () -> Unit,
    modifier: Modifier = Modifier
) {
    LazyColumn(
        modifier = modifier,
        verticalArrangement = Arrangement.spacedBy(AppSpacing.Medium.dp)
    ) {
        // Newsletter Header
        item {
            AppCard(
                modifier = Modifier.fillMaxWidth()
            ) {
                Column(
                    modifier = Modifier.padding(AppSpacing.Medium.dp)
                ) {
                    Row(
                        modifier = Modifier.fillMaxWidth(),
                        horizontalArrangement = Arrangement.SpaceBetween,
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Column(
                            modifier = Modifier.weight(1f)
                        ) {
                            Text(
                                text = newsletter.title,
                                style = MaterialTheme.typography.headlineMedium,
                                color = MaterialTheme.colorScheme.onSurface
                            )
                            
                            Text(
                                text = newsletter.author,
                                style = MaterialTheme.typography.bodyMedium,
                                color = MaterialTheme.colorScheme.onSurfaceVariant
                            )
                        }
                        
                        AppIcon(
                            name = newsletter.icon,
                            size = AppIcon.Large,
                            tint = MaterialTheme.colorScheme.primary
                        )
                    }
                    
                    Spacer(modifier = Modifier.height(AppSpacing.Medium.dp))
                    
                    Text(
                        text = newsletter.description,
                        style = MaterialTheme.typography.bodyLarge,
                        color = MaterialTheme.colorScheme.onSurfaceVariant
                    )
                    
                    Spacer(modifier = Modifier.height(AppSpacing.Large.dp))
                    
                    // Subscribe/Unsubscribe Button
                    AppButton(
                        onClick = onSubscribeClick,
                        variant = if (newsletter.isSubscribed) {
                            ButtonVariant.Outlined
                        } else {
                            ButtonVariant.Primary
                        },
                        text = if (newsletter.isSubscribed) {
                            "Unsubscribe"
                        } else {
                            "Subscribe"
                        },
                        modifier = Modifier.fillMaxWidth()
                    )
                }
            }
        }
        
        // Analytics Section
        item {
            AppCard(
                modifier = Modifier.fillMaxWidth()
            ) {
                Column(
                    modifier = Modifier.padding(AppSpacing.Medium.dp)
                ) {
                    Row(
                        modifier = Modifier.fillMaxWidth(),
                        horizontalArrangement = Arrangement.SpaceBetween,
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Text(
                            text = "Analytics",
                            style = MaterialTheme.typography.titleLarge,
                            color = MaterialTheme.colorScheme.onSurface
                        )
                        
                        AppButton(
                            onClick = onAnalyticsClick,
                            variant = ButtonVariant.Text,
                            text = "View Details"
                        )
                    }
                    
                    Spacer(modifier = Modifier.height(AppSpacing.Medium.dp))
                    
                    // Analytics Chips
                    Row(
                        modifier = Modifier.fillMaxWidth(),
                        horizontalArrangement = Arrangement.spacedBy(AppSpacing.Small.dp)
                    ) {
                        AnalyticsChip(
                            label = "Subscribers",
                            value = newsletter.subscribers.toString(),
                            icon = AppIcon.IconProfile,
                            modifier = Modifier.weight(1f)
                        )
                        
                        AnalyticsChip(
                            label = "Open Rate",
                            value = "${newsletter.openRate}%",
                            icon = AppIcon.IconAnalytics,
                            modifier = Modifier.weight(1f)
                        )
                    }
                    
                    Spacer(modifier = Modifier.height(AppSpacing.Small.dp))
                    
                    Row(
                        modifier = Modifier.fillMaxWidth(),
                        horizontalArrangement = Arrangement.spacedBy(AppSpacing.Small.dp)
                    ) {
                        AnalyticsChip(
                            label = "Click Rate",
                            value = "${newsletter.clickRate}%",
                            icon = AppIcon.IconAnalytics,
                            modifier = Modifier.weight(1f)
                        )
                        
                        AnalyticsChip(
                            label = "Frequency",
                            value = newsletter.frequency,
                            icon = AppIcon.IconCalendar,
                            modifier = Modifier.weight(1f)
                        )
                    }
                }
            }
        }
        
        // Performance Chart
        item {
            PerformanceChart(
                title = "Performance Over Time",
                data = newsletter.performanceData,
                chartType = ChartType.Line,
                modifier = Modifier.fillMaxWidth()
            )
        }
        
        // Preferences Section
        item {
            AppCard(
                modifier = Modifier.fillMaxWidth()
            ) {
                Row(
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(AppSpacing.Medium.dp),
                    horizontalArrangement = Arrangement.SpaceBetween,
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Column(
                        modifier = Modifier.weight(1f)
                    ) {
                        Text(
                            text = "Preferences",
                            style = MaterialTheme.typography.titleMedium,
                            color = MaterialTheme.colorScheme.onSurface
                        )
                        
                        Text(
                            text = "Customize your subscription settings",
                            style = MaterialTheme.typography.bodySmall,
                            color = MaterialTheme.colorScheme.onSurfaceVariant
                        )
                    }
                    
                    AppButton(
                        onClick = onPreferencesClick,
                        variant = ButtonVariant.Outlined,
                        text = "Manage"
                    )
                }
            }
        }
    }
}

@Composable
fun AnalyticsChip(
    label: String,
    value: String,
    icon: String,
    modifier: Modifier = Modifier
) {
    Surface(
        modifier = modifier,
        shape = MaterialTheme.shapes.small,
        color = MaterialTheme.colorScheme.surfaceVariant
    ) {
        Row(
            modifier = Modifier.padding(
                horizontal = AppSpacing.Small.dp,
                vertical = AppSpacing.XSmall.dp
            ),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.spacedBy(AppSpacing.XSmall.dp)
        ) {
            AppIcon(
                name = icon,
                size = AppIcon.Small,
                tint = MaterialTheme.colorScheme.primary
            )
            
            Column {
                Text(
                    text = value,
                    style = MaterialTheme.typography.labelMedium,
                    color = MaterialTheme.colorScheme.onSurface
                )
                
                Text(
                    text = label,
                    style = MaterialTheme.typography.labelSmall,
                    color = MaterialTheme.colorScheme.onSurfaceVariant
                )
            }
        }
    }
}

@Composable
fun PerformanceChart(
    title: String,
    data: List<NewsletterDetail.PerformancePoint>,
    chartType: ChartType,
    modifier: Modifier = Modifier
) {
    AppCard(
        modifier = modifier
    ) {
        Column(
            modifier = Modifier.padding(AppSpacing.Medium.dp)
        ) {
            Text(
                text = title,
                style = MaterialTheme.typography.titleMedium,
                color = MaterialTheme.colorScheme.onSurface,
                modifier = Modifier.padding(bottom = AppSpacing.Medium.dp)
            )
            
            // Chart implementation would go here
            // This is a placeholder for the actual chart component
            Box(
                modifier = Modifier
                    .fillMaxWidth()
                    .height(AppMetrics.ChartMinHeight.dp)
                    .background(
                        color = MaterialTheme.colorScheme.surfaceVariant,
                        shape = MaterialTheme.shapes.medium
                    ),
                contentAlignment = Alignment.Center
            ) {
                Text(
                    text = "Chart: $title (${data.size} points)",
                    style = MaterialTheme.typography.bodyMedium,
                    color = MaterialTheme.colorScheme.onSurfaceVariant
                )
            }
        }
    }
}

@Composable
fun PreferencesSheet(
    newsletter: NewsletterDetail?,
    onDismiss: () -> Unit,
    onSave: (NewsletterDetail.Preferences) -> Unit
) {
    if (newsletter == null) return
    
    var frequency by remember { mutableStateOf(newsletter.preferences.frequency) }
    var notifications by remember { mutableStateOf(newsletter.preferences.notifications) }
    var categories by remember { mutableStateOf(newsletter.preferences.categories) }
    
    AlertDialog(
        onDismissRequest = onDismiss,
        title = {
            Text(
                text = "Subscription Preferences",
                style = MaterialTheme.typography.titleLarge
            )
        },
        text = {
            Column(
                modifier = Modifier.fillMaxWidth(),
                verticalArrangement = Arrangement.spacedBy(AppSpacing.Medium.dp)
            ) {
                // Frequency
                Text(
                    text = "Frequency",
                    style = MaterialTheme.typography.labelMedium,
                    color = MaterialTheme.colorScheme.onSurface
                )
                
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.spacedBy(AppSpacing.Small.dp)
                ) {
                    FilterChip(
                        selected = frequency == "daily",
                        onClick = { frequency = "daily" },
                        label = { Text("Daily") }
                    )
                    
                    FilterChip(
                        selected = frequency == "weekly",
                        onClick = { frequency = "weekly" },
                        label = { Text("Weekly") }
                    )
                    
                    FilterChip(
                        selected = frequency == "monthly",
                        onClick = { frequency = "monthly" },
                        label = { Text("Monthly") }
                    )
                }
                
                // Notifications
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceBetween,
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Text(
                        text = "Notifications",
                        style = MaterialTheme.typography.labelMedium,
                        color = MaterialTheme.colorScheme.onSurface
                    )
                    
                    Switch(
                        checked = notifications,
                        onCheckedChange = { notifications = it }
                    )
                }
                
                // Categories
                Text(
                    text = "Categories",
                    style = MaterialTheme.typography.labelMedium,
                    color = MaterialTheme.colorScheme.onSurface
                )
                
                Text(
                    text = "Select categories to follow",
                    style = MaterialTheme.typography.bodySmall,
                    color = MaterialTheme.colorScheme.onSurfaceVariant
                )
            }
        },
        confirmButton = {
            AppButton(
                onClick = {
                    val newPreferences = NewsletterDetail.Preferences(
                        frequency = frequency,
                        notifications = notifications,
                        categories = categories
                    )
                    onSave(newPreferences)
                },
                variant = ButtonVariant.Primary,
                text = "Save"
            )
        },
        dismissButton = {
            AppButton(
                onClick = onDismiss,
                variant = ButtonVariant.Text,
                text = "Cancel"
            )
        }
    )
}

enum class ChartType {
    Line, Bar, Area, Pie
}

@Composable
fun NewsletterDetailScreenPreview() {
    PiperNewsletterTheme {
        Surface {
            NewsletterDetailContent(
                newsletter = NewsletterDetail(
                    id = "1",
                    title = "Tech Weekly",
                    author = "Tech News Team",
                    description = "Your weekly dose of technology news and insights",
                    icon = AppIcon.IconNewsletter,
                    isSubscribed = true,
                    subscribers = 12500,
                    openRate = 42.5,
                    clickRate = 18.3,
                    frequency = "Weekly",
                    performanceData = emptyList(),
                    preferences = NewsletterDetail.Preferences(
                        frequency = "weekly",
                        notifications = true,
                        categories = listOf("Technology", "Innovation")
                    )
                ),
                onSubscribeClick = {},
                onPreferencesClick = {},
                onAnalyticsClick = {}
            )
        }
    }
}